src/components/PeoplePage/PeoplePage.tsx

```diff
@@ -0,0 +1,78 @@
+/* eslint-disable no-console */
+import { FC, useEffect, useState } from 'react';
+
+import { Loader } from '../Loader';
+import { PeopleTable } from '../PeopleTable';
+import { fetchPeople } from '../../api';
+import { Person } from '../../types';
+
+const getParents = (
+  { motherName, fatherName }: Person,
+  array: Person[],
+) => ({
+  mother: array.find(({ name }) => name === motherName),
+  father: array.find(({ name }) => name === fatherName),
+});
+
+export const PeoplePage: FC = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [hasError, setHasError] = useState(false);
+
+  const isMessageVisible = people.length === 0 && !isLoading;
+  const isListVisible = people.length > 0 && !isLoading;
+
+  const getPeople = async () => {
+    try {
+      const rawPeople = await fetchPeople();
+      const preparedPeople = rawPeople.map(person => ({
+        ...person,
+        ...getParents(person, rawPeople),
+      }));
+
+      setPeople(preparedPeople);
+      setIsLoading(false);
+    } catch (error) {
+      console.error(error);
+
+      setHasError(true);
+    }
+  };
+
+  useEffect(() => {
+    getPeople();
+
+    return () => setPeople([]);
```

Comment:
I did not understand why do you set people to empty array on unmount?

Reply:
So that it reloads on refresh

Reply:
Passing empty array to useEffect accomplish that anyway.  Normally you need to pass abort on unmount but for that you need to create an abortcontroller and pass it as a signal to fetch.

Reply:
Yeah it works the same